enhancements/installer/vsphere-ipi-zonal.md

```diff
@@ -0,0 +1,1067 @@
+	// +kubebuilder:validation:MaxLength=255
+	Server string `json:"server"`
+
+	// port is the TCP port that will be used to communicate to
+	// the vCenter endpoint.
+	// When omitted, this means the user has no opinion and
+	// it is up to the platform to choose a sensible default,
+	// which is subject to change over time.
+	// +kubebuilder:validation:Minimum=1
+	// +kubebuilder:validation:Maximum=32767
+	// +optional
+	Port int `json:"port,omitempty"`
+
+	// The vCenter Datacenters in which the RHCOS
+	// vm guests are located. This field will
+	// be used by the Cloud Controller Manager.
+	// Each datacenter listed here should be used within
+	// a topology.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinItems=1
+	Datacenters []string `json:"datacenters"`
+}
+
+// VSpherePlatformPlacementConstraint is the context information for VM placements within a failure domain
+type VSpherePlatformPlacementConstraint struct {
+	// resourcePool is the absolute path of the resource pool where virtual machines will be
+	// created. The absolute path is of the form /<datacenter>/host/<cluster>/Resources/<resourcepool>.
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=2048
+	// +kubebuilder:validation:Pattern=`^/.*?/host/.*?/Resources.*`
+	// +optional
+	ResourcePool string `json:"resourcePool,omitempty"`
+
+	// folder is the name or inventory path of the folder in which the
+	// virtual machine is created/located.
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=2048
+	// +optional
+	Folder string `json:"folder,omitempty"`
+}
+
+// VSpherePlatformDeploymentSuitable
+// +kubebuilder:validation:Enum=Allowed;NotAllowed
+type VSpherePlatformDeploymentSuitable string
+
+const (
+	// Allowed indicates that the Deployment Zone is suitable for
+	// control plane nodes.
+	Allowed VSpherePlatformDeploymentSuitable = "Allowed"
+
+	// NotAllowed indicates that the Deployment Zone is not suitable for
+	// control plane nodes.
+	NotAllowed VSpherePlatformDeploymentSuitable = "NotAllowed"
+)
+
+// VSpherePlatformDeploymentZone holds the association between a
+// vCenter, failure domain and the virtual machine placementConstraints
+type VSpherePlatformDeploymentZone struct {
+	// name defines the VSpherePlatformDeploymentZoneSpec name.
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=256
+	// +kubebuilder:validation:Required
+	Name string `json:"name"`
+
+	// server is the fully-qualified domain name or the IP address of the vCenter server.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=255
+	Server string `json:"server"`
+
+	// failureDomain is the name of the VSphereFailureDomain used for this VSphereDeploymentZone
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=256
+	FailureDomain string `json:"failureDomain"`
```

Comment:
In how many places are FailureDomains referenced? Can failure domains be used for multiple deployment zones?

Reply:
Right, FailureDomains are all the vCenter locations that RHCOS virtual machines can be provisioned.